### PR TITLE
#506 - Drop database support for PostgreSQL

### PR DESCRIPTION
**Pull request by dncnkrs**
References DbUp/dbup-postgresql#6 

I've been working around this for a while, and I finally had enough time to add this.

I tried to keep everything as similar to the existing code, using the Postgres ensure database and MSSQL drop database methods for reference.

I'm happy to make changes if needed.

### CODE REVIEW

**Comment by knunery** [comment]
This looks good. Can we merge it?

File: src/dbup-postgresql/PostgresqlExtensions.cs [hunk 1]
```diff
@@ -206,6 +182,101 @@ private static void PostgresqlDatabase(this SupportedDatabasesForEnsureDatabase
...
+        }
+
+        if (logger is null)
+        {
+            throw new ArgumentNullException(nameof(logger));
+        }
+
+        var postgresConnectionStringBuilder = new NpgsqlConnectionStringBuilder(connectionString);
+        var database = postgresConnectionStringBuilder.Database;
+        postgresConnectionStringBuilder.Database = "postgres";
```
Comment (Simcon) [inline src/dbup-postgresql/PostgresqlExtensions.cs:265]: This line should be removed since it overwrites the database name in the connection string.

**Comment by Pseudooo** [comment]
Bump?

**Comment by sungam3r** [comment]
@dncnkrs Please rebase onto `master` if you are still interested.

**Pull request closed without merging**